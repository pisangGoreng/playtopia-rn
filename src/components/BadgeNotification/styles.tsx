import {StyleSheet} from 'react-native';
import {COLORS} from '../../utils/colors';
import {horizontalScale, verticalScale} from '../../utils/scaling';

const styles = StyleSheet.create({
  container: {
    height: horizontalScale(32),
    width: horizontalScale(32),
    borderRadius: horizontalScale(32),
    backgroundColor: COLORS.white,
    justifyContent: 'center',
    alignItems: 'center',
  },
  notifContainer: {
    height: horizontalScale(10),
    width: horizontalScale(10),
    borderRadius: horizontalScale(10),
    backgroundColor: COLORS.white,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    right: 0,
  },
  dotContainer: {
    height: horizontalScale(6),
    width: horizontalScale(6),
    borderRadius: horizontalScale(6),
    backgroundColor: COLORS.red,
  },
});

export default styles;
