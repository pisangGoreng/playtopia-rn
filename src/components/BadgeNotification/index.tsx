import React, {useState} from 'react';
import {View} from 'react-native';

import Button from '../Button';
import BellIcon from '../../assets/icons/svg/bell.svg';
import styles from './styles';

const {container, notifContainer, dotContainer} = styles;

const BadgeNotification: React.FC<any> = () => {
  const [haveNotif, setHaveNotif] = useState<boolean>(true);
  const display = haveNotif ? 'flex' : 'none';

  return (
    <Button
      activeOpacity={0.5}
      onPress={() => console.log('notif badge pressed')}>
      <View style={container}>
        <BellIcon />
      </View>

      <View style={[notifContainer, {display}]}>
        <View style={dotContainer} />
      </View>
    </Button>
  );
};

export default BadgeNotification;
