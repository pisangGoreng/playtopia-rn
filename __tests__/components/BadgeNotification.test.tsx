import React from 'react';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';
import BadgeNotification from '../../src/components/BadgeNotification';

describe('Should render a Badge Notification component', () => {
  it('renders the correct compoenent', () => {
    const {toJSON, getByTestId} = render(<BadgeNotification />);

    expect(toJSON()).toMatchSnapshot();

    // const badgeContainer = getByTestId('badge-container');
    // expect(badgeContainer).toBeTruthy();
  });
});
