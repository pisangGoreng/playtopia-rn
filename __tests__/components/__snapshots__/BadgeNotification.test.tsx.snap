// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Badge Notification component renders the correct compoenent 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "red",
        "height": 24,
        "width": 24,
      },
      {
        "borderRadius": 12,
      },
    ]
  }
  testID="badge-container"
>
  <Text>
    Children
  </Text>
</View>
`;

exports[`Should render a Badge Notification component renders the correct message 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "red",
        "height": 24,
        "width": 24,
      },
      {
        "borderRadius": 12,
      },
    ]
  }
  testID="badge-container"
>
  <Text>
    Children
  </Text>
</View>
`;

exports[`Should render a Badge component renders the correct message 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "red",
        "height": 24,
        "width": 24,
      },
      {
        "borderRadius": 12,
      },
    ]
  }
  testID="badge-container"
>
  <Text>
    Children
  </Text>
</View>
`;
